### ISSUE TITLE
Can't deploy Linkerd v2.13.x with ArgoCD

### ISSUE THREAD

**Issue by SansDK** [linkerd/linkerd2#10762]
### What is the issue?

A Kubernetes Lease object is part of the Linkerd helm chart. This type of object is excluded by Argo, and thus the Lease is never created. This causes the `destination` pods to fail.

I think that the Kubernetes lease shouldn't be part of the helm chart, but rather be created at runtime (similar to cert-manager for example). This way, Argo users can also enjoy Linkerd 2.13 :)

Please see this support ticket for more information: https://linkerd.buoyant.io/t/linkerd-destination-pod-unable-to-start-when-using-argocd-since-linkerd-v2-13-0/87/6

### How can it be reproduced?

1. Run ArgoCD in your cluster
2. Try to deploy Linkerd v2.13.0 or v2.13.1

### Logs, error output, etc

```
Error: failed to get lease: ApiError: leases.coordination.k8s.io "policy-controller-write" not found: NotFound (ErrorResponse { status: "Failure", message: "leases.coordination.k8s.io \"policy-controller-write\" not found", reason: "NotFound", code: 404 })


Caused by:
 0: ApiError: leases.coordination.k8s.io "policy-controller-write" not found: NotFound (ErrorResponse { status: "Failure", message: "leases.coordination.k8s.io \"policy-controller-write\" not found", reason: "NotFound", code: 404 })
 1: leases.coordination.k8s.io "policy-controller-write" not found: NotFound
```

### output of `linkerd check -o short`

N/A

### Environment

- Kubernetes v1.25
- Cluster env: AKS
- Linkerd version: v2.13.0
- Argo version: v2.6.7

### Would you like to work on fixing this bug?

None

**Comment by janlonsetteig** [comment]
We are also blocked by this issue. I agree that the Lease object is a runtime object and should not be part of the Helm chart

**Comment by pdeva** [comment]
+1 this is a major blocker

**Comment by anasalsalah** [comment]
+1 please get this fixed ASAP, with thanks.

**Comment by cparadal** [comment]
+1 major blocker for us too. Can't upgrade until this is fixed.

**Comment by risingspiral** [comment]
Thanks for bringing this to our attention. We're currently working on a fix to address this.

**Comment by james-callahan** [comment]
Please don't remove this; it's correct to include it as part of the chart

**Comment by olix0r** [comment]
@james-callahan I agree that we shouldn't need to create Leases at runtime, but it's also not really tenable to prevent all Argo users from upgrading Linkerd. We're working around this in the short term (in #10823), but I continue to hope that Argo will change their behavior so we can revert the change at some point.

**Comment by irizzant** [comment]
As a temp workaround waiting for a new Linkerd release, one can manually add this resource (change the namespace before creating):
```
kind: Lease
metadata:
  name: policy-controller-write
  namespace: linkerd
  labels:
    linkerd.io/control-plane-component: destination
    linkerd.io/control-plane-ns: linkerd
```

to the namespace where Linkerd is deployed, then restart linkerd-destination Pods.

The ArgoCD warning stays there but at least linkerd Pods starts correctly.

**Comment by adleong** [comment]
Thanks to everyone who reported this.  A fix for this has been released in [edge-23.4.3](https://github.com/linkerd/linkerd2/releases/tag/edge-23.4.3).  We'd really appreciate anyone who can give this edge a try and let us know if it resolves the issue for you!

**Comment by SansDK** [comment]
@adleong we've also deployed the `edge-23.4.3` release, and I can also confirm it is working as expected :)
Thanks to the Linkerd team for the quick fix!
